--- pcre/pcre.h	2006-07-05 13:28:01.000000000 +0200
+++ pcre/pcre.h	2006-07-05 13:28:01.000000000 +0200
@@ -60,6 +60,8 @@
 /* Win32 uses DLL by default; it needs special stuff for exported functions
 when building PCRE. */
 
+/* But don't do that when building as part of GLib */
+#if 0
 #ifdef _WIN32
 #  ifdef PCRE_DEFINITION
 #    ifdef DLL_EXPORT
@@ -71,6 +71,7 @@
 #    endif
 #  endif
 #endif
+#endif
 
 /* Otherwise, we use the standard "extern". */
 
